### PR TITLE
Add docs for default encoding, discriminator or content type negotiation

### LINKED ISSUE #1920: Document default interpretation of http layer
The http library assume a lot of defaults for encoding, status code etc. I don't believe we have a documenation on that

## Encoding
- `bytes` does base64 when inside of json payload
- `utcDatetime` and `offsetDateTime`, use `rfc3339` unless in header and then use `rfc7231`
- `duration` use `iso8601`

## Other
- default  status code is: 
  - `400-5xx` if `@error`
  - `200` otherwise

### LINKED ISSUE #1960: Need usage document on discriminators
The only documentation I can find on a search is just the decorator documentation.

We need general documentation on how to model using discriminators, and canonical examples in typespec-azure-core as well.

### PR DESCRIPTION
**Pull request by timotheeguerin**
fix #1960 (Discriminator docs)
fix #1920 (Default interpretations)
partial work for https://github.com/Azure/typespec-azure/issues/2960

## Move discriminator doc to compiler

Moved and expanded the small discriminator doc present in the http library. As discriminator is a compiler decorator and there is nothing specific about http.
https://cadlwebsite.z1.web.core.windows.net/prs/2181/next/standard-library/discriminated-types

## Add documentation on how types are encoded in http 

Describe the default behaviors as well as how to configure
https://cadlwebsite.z1.web.core.windows.net/prs/2181/next/standard-library/http/encoding

## Add content negotiation docs

Add general http docs on how to define different content negotiation
https://cadlwebsite.z1.web.core.windows.net/prs/2181/next/standard-library/http/content-types

### CODE REVIEW

File: docs/standard-library/discriminated-types.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
+# Discriminated types
+
+TypeSpec allows you to represent unions and inheritance however when sending types over the wire many languages need a way to discriminate between the various union variants or model hierarchy.
```
Comment (bterlson) [inline docs/standard-library/discriminated-types.md:3]:
```suggestion
TypeSpec can express unions and inheritance. However, when sending types over the wire many languages need a way to discriminate between the various union variants or models in an inheritance hierarchy.
```

File: docs/standard-library/discriminated-types.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  kind: PetKind.dog;
+  bark: string;
+}
+```
+
+#### Nested discriminator
+
+```tsp
+@discriminator("kind")
+model Pet {
```
Comment (bterlson) [inline docs/standard-library/discriminated-types.md:55]: Shouldn't `Pet` have `kind: string`, and `Cat` have `breed: string`?

File: docs/standard-library/discriminated-types.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+}
+
+@discriminator("breed")
+model Cat extends Pet {
+  kind: "cat";
+  meow: int32;
+}
+
+@discriminator("breed")
+model Siamese extends Pet {
```
Comment (bterlson) [inline docs/standard-library/discriminated-types.md:67]: Maybe this is what you're trying to show but I'm confused why these don't extend `cat`.

File: docs/standard-library/http/encoding.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+<td>
+
+```tsp
+model User {
+  runtime: duration; // ISO8601
+
+  @encode(DurationKnownEncoding.seconds, int32)
+  runtimeInSecondsInt: duration; // in seconds as an int32
+
+  @encode(DurationKnownEncoding.seconds, float32)
```
Comment (bterlson) [inline docs/standard-library/http/encoding.md:108]: I regret having `Known` in these types.

**Review by bterlson — APPROVED** [review]

File: docs/standard-library/discriminated-types.md (hunk 3 above)
Comment (timotheeguerin) [reply docs/standard-library/discriminated-types.md:67]: Nope it’s a mistake

File: docs/standard-library/discriminated-types.md (hunk 2 above)
Comment (timotheeguerin) [reply docs/standard-library/discriminated-types.md:55]: It doesn’t need to but that’s maybe a legacy of not allowing property override. Do you think it should be the recommended way?

File: docs/standard-library/http/encoding.md (hunk 4 above)
Comment (timotheeguerin) [reply docs/standard-library/http/encoding.md:108]: We could remove them and put an Alias. 

Feels like though this makes you think it’s the whole list of values you can use.

**Pull request merged**